### ISSUE TITLE
Crashing on negative fetch refspecs

### ISSUE THREAD

**Issue by lukaspiatkowski** [jesseduffield/lazygit#1967]
**Describe the bug**
Lazygit crashes when opening a git repository containing [negative refspec](https://github.blog/2020-10-19-git-2-29-released/#:~:text=Negative%20refspecs) in a fetch option inside `.git/config`.

**To Reproduce**
1. Clone a repository (e.g. `gh repo clone jesseduffield/lazygit`)
2. Open `.git/config`
3. Add a negative refspec in a `fetch` option under a `[remote ...]` section, e.g.:
```
[remote "origin"]
        url = git@github.com:jesseduffield/lazygit.git
        fetch = +refs/heads/*:refs/remotes/origin/*
        fetch = ^refs/heads/private/*
```
4. Open lazygit inside this repo and see it crashing

**Expected behavior**
Lazygit should operate as usuall and should respect the negative refspec (e.g. without Lazygit if you `git fetch` the branches mentioned by negative refspec will be ignored)

**Screenshots**
<img width="623" alt="Screenshot 2022-05-26 at 13 39 17" src="https://user-images.githubusercontent.com/4467402/170481150-8508412b-0d63-4272-8483-bb50fac800e2.png">

Errors are non-deterministic, once I was able to catch "malformed refspec, separators are wrong":

<img width="671" alt="Screenshot 2022-05-26 at 13 50 31" src="https://user-images.githubusercontent.com/4467402/170482579-511b6c0b-8351-48f1-8351-030b05e6db51.png">

**Desktop (please complete the following information):**
 - OS: MacOs Monterey (Apple Silicon)
 - Lazygit Version v0.34
 - lg --version: `commit=, build date=, build source=homebrew, version=0.34, os=darwin, arch=arm64`

**Additional context**
The stderror message changes after few attempts of running Lazygit in the same terminal:
1. panic: fork/exec /opt/homebrew/bin/git: too many open files ([full output](https://gist.github.com/lukaspiatkowski/5c4f9855d6487fe2a08cf32a68cded8a))
2. panic: pipe: too many open files ([full output](https://gist.github.com/lukaspiatkowski/b64fe1f1123aa0ce380299f500393bd1))
3. panic: open /dev/null: too many open files ([full output](https://gist.github.com/lukaspiatkowski/390d2c52eb3c6f59b1639aff3a0680f4))
It is also hard to catch the screenshot attached above, because lazygit usually clears the screen after this error, but from time to time after few attempts the terminal is so broken that lazygit doesn't clear the screen and other rendering errors happen.

I've tried this with `git.autoFetch = false` in my `~/.config/lazygit/config.yml`, but that didn't help.

**Comment by mark2185** [comment]
Good catch, is this something that stops you from using `lazygit` or did you just notice it by accident? 

I.e. if it's a priority, I can take a look at it today.

**Comment by lukaspiatkowski** [comment]
I've caught it when trying to use negative refspecs to filter out branches that Github's [Merge Queue](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) leaves behind on the main branch. I would say it blocks me from using negative refspecs more than it blocks me from using lazygit for now, but I wanted to merge the negative refspecs for my organization to benefit, so it would be nice to get this sorted sooner rather than later :).

I've tried to quickly fix it now, but I don't know GO and the problem seems to originate in `go-git`, so I don't know if I am doing the right things here. I've managed to patch it up with [this](https://github.com/lukaspiatkowski/lazygit/commit/b47c31b53e4798db94e7cb6bbffe825e7c8c0389 ) commit and it seems to work, but `./test.sh` seems to be stuck:
```
$❯ ./test.sh
ok  	github.com/jesseduffield/lazygit/pkg/tasks	0.329s	coverage: 58.4% of statements
ok  	github.com/jesseduffield/lazygit/pkg/app	0.218s	coverage: 8.2% of statements
?   	github.com/jesseduffield/lazygit/pkg/app/daemon	[no test files]
?   	github.com/jesseduffield/lazygit/pkg/test	[no test files]
?   	github.com/jesseduffield/lazygit/pkg/config	[no test files]
?   	github.com/jesseduffield/lazygit/pkg/integration	[no test files]
?   	github.com/jesseduffield/lazygit/pkg/constants	[no test files]
ok  	github.com/jesseduffield/lazygit/pkg/utils	0.232s	coverage: 44.8% of statements
?   	github.com/jesseduffield/lazygit/pkg/updates	[no test files]
?   	github.com/jesseduffield/lazygit/pkg/env	[no test files]
?   	github.com/jesseduffield/lazygit/pkg/common	[no test files]
?   	github.com/jesseduffield/lazygit/pkg/logs	[no test files]
?   	github.com/jesseduffield/lazygit/pkg/theme	[no test files]
```

I will leave it to you to decide if this PR should go to go-git or jesseduffield's fork and what to do with it.

**Comment by lukaspiatkowski** [comment]
I forgot to mention it here, but I've created an issue: https://github.com/jesseduffield/go-git/pull/1

**Comment by stefanhaller** [comment]
Looks like this was fixed by #2232; closing.
